### PR TITLE
Signup signin

### PR DESCRIPTION
**Pull request by sethcrist**
https://trello.com/c/zia1FFt2/9-react-sign-up-sign-in-log-out

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,26 +1,106 @@
...
   const [apartments, setApartments] = useState(mockApartments)
+  const [currentUser, setCurrentUser] = useState(null)
+  console.log(currentUser)
+
+  // useEffect(() => {
+  //   const loggedInUser = localStorage.getItem("currentUser")
+  //   if (loggedInUser) {
+  //     setCurrentUser(JSON.parse(loggedInUser))
+  //   }
+  // }, [])
```
Comment (sjproctor) [inline src/App.js]: The useEffect can be commented back in and you can remove the log for currentUser.

File: src/App.js [hunk 2]
```diff
@@ -1,26 +1,106 @@
...
+          "Content-Type": "application/json",
+          Accept: "application/json",
+        },
+        body: JSON.stringify(user),
+      })
+      if (!signInResponse) {
+        throw new Error(signInResponse.errors)
+      }
+      const payload = await signInResponse.json()
+      console.log(payload)
```
Comment (sjproctor) [inline src/App.js]: This log can be removed too.

File: src/App.js [hunk 3]
```diff
@@ -1,26 +1,106 @@
...
+          "Content-Type": "application/json",
+          Accept: "application/json",
+        },
+        body: JSON.stringify(currentUser),
+      })
+      if (!signUpResponse) {
+        throw new Error(signUpResponse.errors)
+      }
+      const payload = await signUpResponse.json()
+      console.log(payload)
```
Comment (sjproctor) [inline src/App.js]: Same with this log.

File: src/components/Header.js [hunk 4]
```diff
@@ -20,24 +20,45 @@ const Header = () => {
...
+            tag={Link}
+            to="/signup"
+            className="nav-link"
+            style={{
+              borderRadius: "20px",
+              padding: "10px 20px",
+              backgroundColor: "#CBA86B",
+              color: "black",
+              border: "none",
+            }}
```
Comment (sjproctor) [inline src/components/Header.js:39]: I know we talked about this button with the Link tag before, and I am pretty sure this won't work. If it does, great! But I would double check.

File: src/pages/SignUp.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        </FormGroup>
+        <FormGroup>
+          <Label for="Password_Confirmation">Password Confirmation</Label>
+          <input
+            id="Password_Confirmation"
+            name="Password_Confirmation"
+            placeholder="Password Confirmation"
+            type="Password"
+            className="form-control"
+            {...register("Password_Confirmation", { required: true })}
```
Comment (sjproctor) [inline src/pages/SignUp.js]: I would rename the label, id, name, and register to be lowercase `password_confirmation`
Also your errors has enjoys on it.

**Review by sjproctor — COMMENTED** [review]
Few things to clean up.

File: src/components/Header.js (hunk 4 above)
Comment (sethcrist) [reply src/components/Header.js:39]: when we look at the application tab/ local storage it deletes the user token. Even with that link all the buttons work.

**Review by sjproctor — APPROVED** [review]

**Pull request merged**